src/FlowtideDotNet.Substrait/Sql/Internal/SqlSubstraitVisitor.cs

```diff
@@ -460,19 +461,134 @@ private RelationData VisitSelectAggregate(Select select, ContainsAggregateVisito
 
         protected override RelationData? VisitTableWithJoins(TableWithJoins tableWithJoins, object? state)
         {
-            var parent = Visit(tableWithJoins.Relation!, state);
+            RelationData? parent = null;
+            if (IsTableFunction(tableWithJoins.Relation))
+            {
+                parent = VisitTableFunctionRoot(tableWithJoins.Relation);
+            }
+            else
+            {
+                parent = Visit(tableWithJoins.Relation!, state);
+            }
+            
             Debug.Assert(parent != null);
             if (tableWithJoins.Joins != null)
             {
                 foreach (var join in tableWithJoins.Joins)
                 {
+                    if (IsTableFunction(join.Relation))
+                    {
+                        return VisitTableFunctionJoin(join, parent);
+                    }
                     parent = VisitJoin(join, parent, state);
                     Debug.Assert(parent != null);
                 }
             }
             return parent;
         }
 
+        private static bool IsTableFunction(TableFactor? tableFactor)
+        {
+            if (tableFactor is TableFactor.Table table &&
+                table.Args != null)
+            {
+                return true;
+            }
+            return false;
+        }
```

Comment:
```suggestion
        private static bool IsTableFunction(TableFactor? tableFactor) 
              => tableFactor is TableFactor.Table table && table.Args != null;
```

Could be shortened a bit.

Reply:
Could maybe also be extracted into an extension method on `TableFactor` since the same check is also used here:
https://github.com/koralium/flowtide/pull/450/files#diff-6cc00e6573b10e4b46945fd836aff8f21d56e46c2f4e1cacec1917b24becf5fbR502

```csharp
public static TableFactorExtensions
{
        public static bool IsTableFunction(this TableFactor? tableFactor) 
              => tableFactor is TableFactor.Table table && table.Args != null;
}
```

Reply:
Used the first suggestion, in the other case it is used to access the args